torch/distributed/device_mesh.py

```diff
@@ -254,6 +255,9 @@ def __init__(
                     self._get_or_create_default_group()
                     self._init_process_groups()
 
+                if get_backend() == "threaded":
```

Comment:
Moving this check out of `if _init_backend:` check since, even if _init_backend()==False, default backend would be initialized at this point. 

Without this, the DeviceMesh from group tests (MTPG) would fail, because _init_backend==False.